lib/plugins/aws/lib/naming.test.js

```diff
@@ -463,4 +470,16 @@ describe('#naming()', () => {
           .to.equal('FunctionNameLambdaPermissionLogsSubscriptionFilterCloudWatchLog0');
       });
   });
+
+  describe('#getLambdaCognitoUserPoolPermissionLogicalId()', () => {
+    it('should normalize the function name and add the standard suffix including the normalized ' +
+      ' pool name and trigger source',
+      () => {
+        expect(sdk.naming.getLambdaCognitoUserPoolPermissionLogicalId(
+          'functionName',
+          'poolId',
+          'TriggerSource'
+        )).to.equal('FunctionNameLambdaPermissionCognitoUserPoolTriggerSourcePoolIdTriggerSource');
```

Comment:
Can't seem to find any information regarding length limits, the only thing I can find is [this](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html).